src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -107,12 +109,12 @@ public String getApiUri() {
         StandardCredentials credentials = Connector.lookupScanCredentials(observer.getContext(), apiUri, scanCredentialsId);
         GitHub github = Connector.connect(apiUri, credentials);
         if (credentials != null && !github.isCredentialValid()) {
-            listener.getLogger().format("Invalid scan credentials, skipping%n");
+            listener.getLogger().format("Invalid scan credentials %s to connect to %s, skipping%n", CredentialsNameProvider.name(credentials), apiUri == null ? "github.com" : apiUri);
             return;
         }
 
         if (!github.isAnonymous()) {
-            listener.getLogger().format("Connecting to GitHub using %s%n", CredentialsNameProvider.name(credentials));
+            listener.getLogger().format("Connecting to %s using %s%n", apiUri == null ? "github.com" : apiUri, CredentialsNameProvider.name(credentials));
```

Comment:
@recena `RepositoryUriResolver#hostnameFromApiUri` just returns the hostname when I would like to display the entire API URL. I suspect that some people will make a mistake defining the API URL and will be happy to see the entire URL.

BTW I did not catch why `GitHubSCMNavigator#apiUri` is nullable with a kind of discovery happening latter on.